### PR TITLE
Asset building: replace laravel-mix with esbuild

### PR DESCRIPTION
**Pull request by verheyenkoen**

### CODE REVIEW

File: cli/server_cmd.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ var serverCmd = &cobra.Command{
 
 		// setup assets
 		assets, err := mix.New(mix.Config{
-			ManifestFile: "static/mix-manifest.json",
+			ManifestFile: "static/manifest.json",
 			PublicPath:   "/static/",
 		})
```
Comment (verheyenkoen) [inline cli/server_cmd.go:71]: @nics @netsensei Do we still need to keep using the dependency on https://github.com/ugent-library/mix? Kinda weird if we're not using Laravel Mix anymore. In the end it's just loading a JSON file and keeping it as a map for translating assets to their destination path (including md5 hash). Or maybe the manifest file is not even necessary and we just translate the asset path in Go-code and calculate the hash there.

**Comment by verheyenkoen** [comment]
@mietcls Can you review the changed scss files from this PR plz? Does this include any files that you would overwrite when applying theme updates?

Also currently there's one deprecation warning during build because a dependency from bootstrap uses the `abs()` function with a percentage. This has already been fixed in the latest bootstrap version so this would probably disappear on the next update.

File: cli/server_cmd.go (hunk 1 above)
Comment (nics) [reply cli/server_cmd.go:71]: no but we need a replacement asset path helper
calculating all the hashes on boot you mean? that seems like unnecessary bookkeeping, when we can just read the manifest?

Comment (verheyenkoen) [reply cli/server_cmd.go:71]: I would say: calculate every hash on first use and cache it from there. Also, I had to manually implement the hash calculation in the manifest myself this time as esbuild does this differently. That's kind of the only added value of the manifest for now I guess.

File: cli/server_cmd.go [hunk 2]
```diff
@@ -233,3 +232,17 @@ var serverCmd = &cobra.Command{
...
+		return nil, fmt.Errorf("couldn't read manifest '%s': %w", manifestFile, err)
+	}
+
+	assets := make(map[string]string)
+	if err = json.Unmarshal(data, &assets); err != nil {
+		return nil, fmt.Errorf("couldn't parse manifest '%s': %w", manifestFile, err)
+	}
+
+	return assets, nil
+}
```
Comment (verheyenkoen) [inline cli/server_cmd.go:248]: This function might need to be placed somewhere else but didn't really know where.

**Comment by verheyenkoen** [comment]
Some manual sass fixes need to be backported to the prototype repo.

File: cli/server_cmd.go (hunk 1 above)
Comment (verheyenkoen) [reply cli/server_cmd.go]: I managed to get the esbuild built-in hashes (in filename) working for te fonts referenced in sass now so calculating md5 hashes ourselves is no longer relevant.

**Comment by verheyenkoen** [comment]
Also sent images through ImageOptim which reduced the buindle size with about 200Kb (20%).

File: reflex.conf.example [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # restart app server when a .go file changes
--s -r '\.go$' -- sh -c 'go run main.go server'
+-s -r '(\.go|manifest.json)$' -- sh -c 'go run main.go server'
```
Comment (nics) [inline reflex.conf.example:2]: @verheyenkoen this is a better regex: `\.go$|^manifest.json$`, no need for capturing

File: cli/server_cmd.go (hunk 2 above)
Comment (nics) [reply cli/server_cmd.go:248]: there's more stuff that needs a place outside the cli package in an application builder, we can move all it in a later refactor

**Review by nics — CHANGES_REQUESTED** [review]

**Review by nics — APPROVED** [review]

**Pull request merged**